src/game.cpp

```diff
@@ -1171,24 +1171,36 @@ bool game::cleanup_at_end()
         if( characters.empty() ) {
             bool queryDelete = false;
             bool queryReset = false;
-
-            if( get_option<std::string>( "WORLD_END" ) == "query" ) {
-                uilist smenu;
-                smenu.allow_cancel = false;
-                smenu.addentry( 0, true, 'k', "%s", _( "Keep world" ) );
-                smenu.addentry( 1, true, 'r', "%s", _( "Reset world" ) );
-                smenu.addentry( 2, true, 'd', "%s", _( "Delete world" ) );
-                smenu.query();
-
-                switch( smenu.ret ) {
-                    case 0:
-                        break;
-                    case 1:
-                        queryReset = true;
-                        break;
-                    case 2:
-                        queryDelete = true;
-                        break;
+            bool decided = false;
+            std::string buffer = _( "Warning, Cataclysm: Dark Days Ahead is not really made for "
+                    "multiple runs in a single world.  "
+                    "You will experience some bugs if you have already played extensively in "
+                    "this world.\n\n"
+                    "Are you sure you wish to keep this world?"
+                    );
+
+            while( !decided ) {
```

Comment:
This will cause an infinite loop if the WORLD_END option is anything but "query". In that case, the loop body is effectively nothing and `decided` will never change.

I suggest you move the check for the WORLD_END option outside of the loop. And move `decided` and `buffer` inside the if branch.